Source/WebCore/css/StyleProperties.cpp

```diff
@@ -690,6 +692,45 @@ String StyleProperties::fontVariantValue() const
     return result.toString();
 }
 
+String StyleProperties::fontSynthesisValue() const
+{
+    StringBuilder result;
+
+    auto getLonghandValue = [&](CSSPropertyID propertyID) {
+        auto foundPropertyIndex = findPropertyIndex(propertyID);
+        if (foundPropertyIndex == -1)
+            return String();
+
+        auto property = propertyAt(foundPropertyIndex);
+        return property.value()->cssText();
+    };
+
+    auto appendWithPrefixIfNeeded = [&](const char* word) {
+        if (!result.isEmpty())
+            result.append(' ');
+        result.append(String(word, strlen(word)));
+    };
+    
+    // font-synthesis: none | [ weight || style || small-caps ]
+    auto weightValue = getLonghandValue(CSSPropertyFontSynthesisWeight);
+    auto styleValue = getLonghandValue(CSSPropertyFontSynthesisStyle);
+    auto capsValue = getLonghandValue(CSSPropertyFontSynthesisSmallCaps);
+
+    // font-synthesis: none or a CSS wide value keyword
+    if (weightValue == styleValue && styleValue == capsValue)
+        return weightValue;
```

Comment:
these are the values for the short-hand, so they can only be  none | [ weight || style || small-caps ]

Reply:
IIUC, if you have all 3 longhands set to auto, and `weightValue == styleValue && styleValue == capsValue` would return `true`, so it would return `font-synthesis: auto` instead of `font-synthesis: weight style small-caps`. That seems wrong to me, since `font-synthesis: auto` doesn't exist.

Reply:
you are right, fixing it!